frontend/src/data/queries/query.spec.ts

```diff
@@ -147,6 +147,32 @@ describe('QuerySet compile method', () => {
         expect(out[0].has('pets:cat')).toEqual(true)
     })
 
+    test('1 category 1 dimension with category dependency ', () => {
+        const A = new Query('pets', 'cat')
+        const qs = new QuerySet(categoryDimensions, [['pets', 'tools']], A)
+        const out = qs.compile()
+        expect(out.length).toEqual(2)
+        expect(out[0].has('pets:cat')).toEqual(true)
+        expect(out[0].has('tools:hammer')).toEqual(true)
+        expect(out[1].has('pets:cat')).toEqual(true)
+        expect(out[1].has('tools:ruler')).toEqual(true)
+    })
+
+    test('2 categories, single dimension category has a dependency on unqueried category ', () => {
```

Comment:
I thought this logic was already merged into `main`

Reply:
Not sure what happened here, mightve messed up the merge, Should be  fixed now